std/traits.d

```diff
@@ -5154,6 +5157,22 @@ unittest
 }
 
 /**
+Detect whether `T` is a scalar type that promotes to an integral type (that is, a built-in
+integral, character, or boolean type).
+ */
+enum bool isFixedPoint(T) = isIntegral!T || isSomeChar!T || isBoolean!T;
```

Comment:
> Interesting rationale, but the converse of floating point is not fixed point...

I am not particularly attached to the `isFixedPoint` name - I'm just waiting for someone to suggest something better: descriptive and meaningfully shorter than the implementation.

> How frequent is this likely to be **[used]**?

When converting functions into templates, people tend to replace the parameter `int a` with `A a` ... `if (isIntegral!A)`. The two are not fully compatible, though, because `int a` will accept `bool`, `char`, and `wchar` as well.

The range of types accepted by `A a` ... `if (isFixedPoint!A)` is much closer to those accepted by `int a`. In my opinion most template functions with `isIntegral` in the constraints are candidates for conversion to `isFixedPoint`, instead.

Of course, the whole thing is just a hack to work around a weakness in the language: the inability to declare a template function in such a way that type deduction works properly when implicit conversions are involved.

> Should it include integral `enum` types?

I considered it unwise to deviate from the behaviour of the existing `isWhatever` traits, so it does the same thing that `isIntegral` does: it gives the same answer for an `enum` type as it would for the `enum`'s base type.